deployment/checkCmd.py

```diff
@@ -2,10 +2,52 @@
 import os
 import logging
 import logging.config
+import yaml
+import re
+from k8sPaiLibrary.maintainlib import common as pai_common
 
 logger = logging.getLogger(__name__)
 
 
+def validate_config(config_path):
+    layout_file = os.path.join(config_path, "layout.yaml")
+    old_cluster_config_file = os.path.join(config_path, "cluster-configuration.yaml")
+    if(os.path.exists(old_cluster_config_file) and not os.path.exists(layout_file)):
+        logger.error("[Error], please upgrade config files!")
+        exit(-1)
+
+    # kubernetes config
+    my_cluster_object_model = cluster_object_model(config_path)
+    kubernetes_config = my_cluster_object_model.kubernetes_config()
+    logger.info("[OK] vaildate kubernetes config.")
+
+    # pai service config
+    service_config = my_cluster_object_model.service_config()
+    logger.info("[OK] vaildate PAI services config.")
+
+
+def validata_node_os(config_path):
+    layout_file = os.path.join(config_path, "layout.yaml")
+    hosts = yaml.load(open(layout_file))["machine-list"]
+    print hosts
+
+    host_not_match = []
+    regex = r"Description:\s+Ubuntu\s+16.04(.\d)?\s+LTS"
+    for host in hosts:
+        logger.info("Check os for %s ...", host["hostip"])
+        result_stdout, result_stderr = pai_common.ssh_shell_paramiko_with_result(
```

Comment:
do not need to use regex here.
you can check file `/etc/os-release`, grep `ID=ubuntu` and `VERSION_ID="16.04"`.

Reply:
Hi @abuccts, 

Agree, using grep also works. The two method are equivalent, so either is fine.